1-Draft/RFCNNNN-Notification-On-Version-Update.md

```diff
@@ -0,0 +1,298 @@
+---
+RFC: RFCXXXX
+Author: Dongbo Wang
+Status: Draft
+SupercededBy: N/A
+Version: 1.0
+Area: Console
+Comments Due: 4/30/2019
+Plan to implement: Yes
+---
+
+# Notification on PowerShell version updates
+
+Today, to find out whether a new version of PowerShell is available,
+one has to check the release page of the `PowerShell\PowerShell` repository,
+or depend on communication channels like `twitter` or `GitHub Notifications`.
+It would be convenient if `pwsh` itself can notify the user of a new update on startup.
+
+## Motivation
+
+    As a PowerShell user, I get notified when a new version of `pwsh` becomes available.
+
+## Specification
+
+### Target Goals
+
+1. No notification or update check when the running `pwsh` is a self-built version.
+No notification or update check for non-interactive sessions.
+Also, no notification when the PowerShell banner message is suppressed.
+
+2. When there is a new update, assuming you use `pwsh` every day
+and at least one interactive `pwsh` session lasts long enough for the update check,
+then you should be able to see an update notification during the `pwsh` startup on the same day of a release or the next day at the latest.
+
+3. This feature must have very minimal impact on the startup time of `pwsh`.
+This means the check for update must not happen during `pwsh` startup.
+The only acceptable extra overhead to the `pwsh` startup should just be the work related to printing the notification.
+
+4. Check for updates should not blindly run for every interactive `pwsh` session.
+For a particular version of `pwsh`, only one check at most can run to complete per day
+no matter how many interactive session of the `pwsh` are started/opened in that day.
+
+5. After a new update is detected during a successful check,
+all subsequent interactive sessions of that version of `pwsh` should show the notification at startup time.
+And subsequent checks can be avoided for a reasonable period of time, such as a week.
+
+6. `pwsh` of preview versions should check for the new preview version as well as the new GA version.
+`pwsh` of GA versions should check for the new GA version only.
+
+7. The notification and update check are not needed in some scenarios,
+such as when `pwsh` is in a container image.
+Hence, you should be able to suppress them altogether by setting an environment variable.
+
+### Non-Goals
+
+1. Notification shows up right after a new version of `pwsh` is released.
+
+   _This is not a goal._
+   Assuming you use `pwsh` interactively every day,
+   then a notification about the new release may show up on the same day,
+   but is guaranteed no later than the next day.
+
+2. If an update check detects a new release, the notification should show up in the same session.
```

Comment:
from an implementation perspective, if this is done as a `Cmdlet` it should be possible for a user to check if a new version is available at any time.